org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/RunnableDeviceServiceImpl.java

```diff
@@ -206,27 +206,27 @@ public final IPositioner createPositioner() throws ScanningException {
 	public final <T> IRunnableDevice<T> createRunnableDevice(T model, IPublisher<ScanBean> publisher) throws ScanningException {
         return createRunnableDevice(model, publisher, true);
 	}
-	
+
 	@Override
 	public final <T> IRunnableDevice<T> createRunnableDevice(T model, IPublisher<ScanBean> publisher, boolean configure) throws ScanningException {
-				
+
 		try {
 			if (deviceConnectorService==null) deviceConnectorService = getDeviceConnector();
-			
+
 			final IRunnableDevice<T> scanner = createDevice(model);
 			if (scanner instanceof AbstractRunnableDevice) {
 				AbstractRunnableDevice<T> ascanner = (AbstractRunnableDevice<T>)scanner;
 				ascanner.setRunnableDeviceService(this);
                 ascanner.setConnectorService(deviceConnectorService);
                 ascanner.setPublisher(publisher); // May be null
-                
+
                 // If the model has a name for the device, we use
                 // it automatically.
                 try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)